### ISSUE TITLE
Unable To Test Fields Bound to Model Properties

### ISSUE THREAD

**Issue by LarsaSolidor** [laravel-filament/filament#5105]
### Package

filament/forms

### Package Version

v2.16.52

### Laravel Version

v9.40.1

### Livewire Version

v2.10.7

### PHP Version

PHP 8.1

### Problem description

When asserting the existance of a field on a form made using dot notation to store the value on a bound model, the test incorrectly fails. 

`TestsForms`
```php 
$field = data_get($form->getFlatFields(withHidden: true), $fieldName, null);
```  
Evaluates to null as a result of `helpers` splitting the field name `user.type_id` into an array of `['user', 'type_id']`, so it does not return the field.

### Expected behavior

The test correctly detects the presence of a field with a name of `user.type_id`.

### Steps to reproduce

Create a form with a field with a dotted fieldname

```php
protected function getFormSchema(): array
{
    [Select::make('user.type_id')]
}
```

Run a test that asserts the presence of that field
```php
Livewire::test(UserForm::class)
    ->assertFormFieldExists('user.type');
```

### Reproduction repository

https://github.com/laravel/laravel

### Relevant log output

```shell
Failed asserting that a field with the name [user.type_id] exists on the form with the name [form] on the [App\Http\Livewire\User\UserForm] component.
```

**Comment by danharrin** [comment]
1) We don't support binding fields to models. We consider this a bad practice, as Filament form data does not always correspond 1:1 with your model attributes. To ensure all Filament fields work correctly, you **must** fill the form using `$this->form->fill()` and get data using `$this->form->getState()`. We suggest that you bind all fields to an array on your Livewire component, and scope them with `statePath()` - <https://filamentphp.com/docs/2.x/forms/getting-started#scoping-form-data-to-an-array-property>.

2) When we ask you for a reproduction repository URL, we expect some effort to be made by the person reporting the issue. In return, we provide you with free support. If you try and dodge the system, like providing a link to the laravel/laravel repository, all you are doing is creating more work for us. We then have to reproduce your issue for you, before we fix it. This wastes time that we could be spending fixing other issues. In the future, if you still don't see the value in submitting a reproduction repository, please create a PR to fix your issue instead.

3) I will investigate to see if this issue still occurs with array properties. If you pass a `statePath()` to scope your form, it will definitely work. But I'm not sure about nesting *within* that array.

**Comment by danharrin** [comment]
Fixed for nested arrays in e2746b9ef1f688f5906633fa9c261e084d0af9f2
